src/assertions/gleu.ts

```diff
@@ -0,0 +1,128 @@
+      const referenceNGramCounts = new Map<string, number>();
+      
+      for (const gram of candidateNGrams) { // Calculate the count of each element in the candidate n-gram string. 
+        candidateNGramCounts.set(gram, (candidateNGramCounts.get(gram) || 0) + 1);
+      }
+      
+      for (const gram of referenceNGrams) { // Calculate the count of each element in the reference n-gram string
+          referenceNGramCounts.set(gram, (referenceNGramCounts.get(gram) || 0) + 1);
+      }
+      
+      // Calculate matching n-grams 
+      for (const [gram, candidateCount] of candidateNGramCounts.entries()) { //
+        const referenceCount = referenceNGramCounts.get(gram) || 0; // calculate the count of matching elements from the candidate n-gram string & the reference n-gram string. 
+        matchCount += Math.min(candidateCount, referenceCount);  
+      }
+      
+      // Update totals elements.
+      candidateTotal += candidateNGrams.length;
+      referenceTotal += referenceNGrams.length;
+    }
+    
+    // Calculate GLEU as (matches) / max(candidate_total, reference_total) to reduce the number of division operations. 
+    const denominator = Math.max(candidateTotal, referenceTotal);
+    return denominator > 0 ? matchCount / denominator : 0;
+  });
+  
+  // Return the maximum score across all references
+  return Math.max(...referenceScores);
+}
+
+/**
+ * Handles GLEU (Google-BLEU) score calculation and evaluation for assertions.
+ * GLEU is a variant of BLEU that correlates better with human judgments on sentence-level evaluation.
+ * 
+ * @param {Object} params - The parameters for GLEU score evaluation
+ * @param {Object} params.assertion - The assertion configuration object
+ * @param {boolean} params.inverse - Whether to invert the pass condition
+ * @param {string} params.outputString - The candidate string to evaluate
+ * @param {string|string[]} params.renderedValue - The reference string(s) to compare against
+ * @param {string} params.provider - The provider name (unused)
+ * @param {Object} params.test - The test case data (unused)
+ * @returns {GradingResult} Object containing:
+ *   - pass: boolean indicating if assertion passed
+ *   - score: normalized GLEU score (0-1)
+ *   - reason: explanation of the result
+ *   - assertion: original assertion config
+ * @throws {Error} If renderedValue is not a string or array of strings
+ */
+
+export function handleGleuScore({
+  assertion,
+  inverse,
+  outputString,
+  renderedValue,
+  provider, // Use if your assertion needs provider-specific logic
+  test,     // Access to test case data
+}: AssertionParams): GradingResult {
+  // Validate inputs
+  invariant(
+    typeof renderedValue === 'string' ||
+    (Array.isArray(renderedValue) && renderedValue.every((v) => typeof v === 'string')),
+    '"gleu" assertion must have a string or array of strings value'
+  );
+
+  const threshold = assertion.threshold ?? 0.5;
```

Comment:
@mldangelo I'm hoping the threshold value is correct. Let me know if not, and I can change it.

Reply:
0.5 is a good choice